xdsl/dialects/builtin.py

```diff
@@ -2229,8 +2237,16 @@ def create_dense_int(
     @staticmethod
     def create_dense_float(
         type: RankedStructure[_FloatAttrType],
-        data: Sequence[float] | Sequence[FloatAttr[_FloatAttrType]],
+        data: float
+        | FloatAttr[_FloatAttrType]
+        | Sequence[float]
+        | Sequence[FloatAttr[_FloatAttrType]],
     ) -> DenseIntOrFPElementsAttr[_FloatAttrType]:
+        if isinstance(data, FloatAttr):
+            data = data.value.data
+        assert not isinstance(data, int)  # WHY???
```

Comment:
I'm curious, do you have an idea why this is needed?
Otherwise, we should remove the comment and add a better one, or remove it

Reply:
I forgot I added the comment, but I really don't know why. Somehow after ruling out `FloatAttr`, pyright decides that `data` could be an `int`. I feel this has come up before and is some weird intersection of the rules for floats including `int`. I'm also pretty sure that this code doesn't work without this line.

Reply:
Ok this is my bad, turns out python lets you pass an int to this function, so pyright was correct

Reply:
Oh yes, isn't something like int being a subset of float?

Reply:
For function calls but not for `isinstance` checks or something?